doc/design.adoc

```diff
@@ -0,0 +1,637 @@
+| string
+| true
+|
+
+| `type`
+| Type of remote site configuration.
+| `Static` or `Dynamic`
+| true
+|
+
+| `host`
+| Remote site host name.
+| string
+| true
+|
+
+| `port`
+| Remote site host port (only for `Static` type).
+| int32
+| false
+|
+
+| `secret`
+| Secret to connect to remote site (only for `Dynamic` type).
+| <<infinispansecretspec>>
+| false
+|
+
+|=======================
+
+
+[[infinispansecretspec]]
+##### `InfinispanSecretSpec`
+
+`InfinispanSecretSpec` defines how `InfinispanSpec` secrets are configured.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required | Default
+
+| `type`
+| Type of secret.
+| `Credentials`, `Keystore` or `Token`
+| true
+|
+
+| `secretName`
+| Name of referenced secret.
+| string
+| true
+|
+
+|=======================
+
+If type is `Credentials`, Secret` is expected to contain username and password credentials.
+These would be defined in `stringData/username` and `stringData/password` fields respectively.
+
+If type is `Keystore`, `Secret` is expected to contain base64 encoded data in `data/keystore.p12` field.
+Optional keystore password would be located in `stringData/password` field.
+
+If type is `Token`, `Secret` is expected to contain base64 encoded data in `stringData/token` field.
+
+
+[[infinispanstatus]]
+### `InfinispanStatus`
+
+`InfinispanStatus` is the most recent observed status of the `InfinispanSpec`. Read-only.
+
+This is a proposal. TODO: Please comment
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `nodes`
+| Status of the nodes.
+| []<<InfinispanNodeStatus>>
+| true
+
+| `summary`
+| Short description of the cluster status. Now summary is `Ready` if all the pod conditions of type `Ready` are true 
+| string
+| true
+
+|=======================
+
+
+[[nodes]]
+#### `InfinispanNodeStatus`
+
+`InfinispanNodeStatus` is the most recent observed status of a pod running `InfinispanSpec`.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `name`
+| Name of the Pod.
+| string
+| true
+
+| `clusterMembers`
+| List of cluster members (in Infinispan CLI format)
+| string
+| true
+
+| `clusterSize`
+| Number of cluster members
+| int
+| true
+
+|=======================
+
+## Full Example
+
+.full-example.yaml
+[source,yaml]
+----
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: full-example-infinispan
+spec:
+  evictionPolicy: Reject
+  image: jboss/infinispan-server:latest
+  replicationFactory: 3
+  size: 4
+  cluster:
+    encryption:
+      type: Symmetric
+      secret:
+        type: Keystore
+        secretName: cluster-encrypt-secret
+    secret:
+      type: Credentials
+      secretName: cluster-auth-secret
+  connector:
+    secret:
+      type: Credentials
+      secretName: connect-auth-secret
+  container:
+    jvmOptionsAppend: "-XX:NativeMemoryTracking=summary"
+    memory: 1Gi
+    storage: 2Gi
```

Comment:
@Crumby `storage` is only relevant to datagrid use case, shall we store it under `datagrid` field? e.g.

```
spec:
  datagrid:
    storage: 2Gi
```

Reply:
`container` suites this option nicely. I wouldn't move it to `datagrid` (and kept datagrid only as an option for `profile`)

Reply:
@Crumby See #77. I made a `spec.datagrid` a subtree since several pieces fit well under it (sites, datasources) and also put `storage` there. I wouldn't mind much to put it back under `spec.container` but this is really a datagrid specific option. Could maybe put it under `spec.datagrid.container.storage` so that it's clearer.... Let me know your thoughts in the new PR.